pkg/controller/executors/kube.go

```diff
@@ -96,7 +104,7 @@ type writeThroughCachingWriter struct {
 }
 
 func IDFromObject(obj client.Object, op string) []byte {
-	return []byte(fmt.Sprintf("%s:%s:%s:%s:%s", obj.GetClusterName(), obj.GetObjectKind().GroupVersionKind().String(), obj.GetNamespace(), obj.GetName(), op))
+	return []byte(fmt.Sprintf("%s:%s:%s:%s:%s", obj.GetGenerateName(), obj.GetObjectKind().GroupVersionKind().String(), obj.GetNamespace(), obj.GetName(), op))
```

Comment:
Is this going to break backwards compatibility if we deploy while a workflow is running?

Reply:
updated it, it should be [GetZZZ_DeprecatedClusterName](https://github.com/kubernetes/apimachinery/blob/97e5df2d0258ac077093867fabf508c1fc31f53b/pkg/apis/meta/v1/meta.go#L175-L177). `GetClusterName()` is remove from latest version of k8s API.

Reply:
On second thought, this is an in-memory cache. There is no need for backwards compatibility, only uniqueness guarantees. Since this is always empty, let's just drop it completely.

Reply:
That makes sense, thank you. removed it